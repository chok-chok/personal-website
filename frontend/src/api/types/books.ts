export interface ListBooksQuery { 
    status?: string
};

export interface Book {
    title: string,
    author: string,
    readStatus: string,
    coverImage: string,
    review?: string,
    rating?: number,
    readDate?: Date,
    id?: string
}

