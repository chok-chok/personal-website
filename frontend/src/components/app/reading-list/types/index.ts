export interface ReadingListHeaderProps {
  headerText: string;
  bodyText: string;
}

export interface RatingProps {
  rating: number;
}

export interface ActivelyReadingBook {
  title: string;
  author: string;
  coverImage: string;
}
