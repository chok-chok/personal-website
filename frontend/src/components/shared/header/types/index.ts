export interface MenuItem {
  title: string;
  url: string;
}

export interface MenuProps {
  menus: MenuItem[];
}
