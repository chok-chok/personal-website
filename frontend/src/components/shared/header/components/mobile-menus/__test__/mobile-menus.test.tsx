import * as React from 'react';
import MobileMenus from '../mobile-menus';
import { shallow } from 'enzyme';
import { MenuProps } from '../../../types';

const props: MenuProps = {
  menus: [
    {
      title: 'Home',
      url: '/',
    },
    {
      title: 'Reading list',
      url: '/reading-list',
    },
    {
      title: 'Contact',
      url: '/contact',
    },
  ],
};

describe('Test MobileMenus component', () => {
  it('should renders correctly', () => {
    const tree = shallow(<MobileMenus menus={props.menus} />);

    expect(tree).toMatchSnapshot();
  });

  it('should renders correct number of menus', () => {
    const wrapper = shallow(<MobileMenus menus={props.menus} />);

    expect(wrapper.find('.mobile-menu')).toHaveLength(props.menus.length);
  });
});
