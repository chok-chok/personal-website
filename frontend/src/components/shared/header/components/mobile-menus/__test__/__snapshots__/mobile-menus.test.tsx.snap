// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MobileMenus component should renders correctly 1`] = `
<Fragment>
  <styled__MobileMenusWrapper>
    <styled__MobileMenuLink
      onClick={[Function]}
    >
      <styled__MobileMenuIcon
        src="/svg/menu.svg"
      />
    </styled__MobileMenuLink>
    <styled__StyledDrawer
      anchor="right"
      onClose={[Function]}
      open={false}
    >
      <styled__StyledMobileMenuItem
        className="mobile-menu"
        href="/"
        key="0"
      >
        Home
      </styled__StyledMobileMenuItem>
      <styled__StyledMobileMenuItem
        className="mobile-menu"
        href="/reading-list"
        key="1"
      >
        Reading list
      </styled__StyledMobileMenuItem>
      <styled__StyledMobileMenuItem
        className="mobile-menu"
        href="/contact"
        key="2"
      >
        Contact
      </styled__StyledMobileMenuItem>
    </styled__StyledDrawer>
  </styled__MobileMenusWrapper>
</Fragment>
`;
